cmd/kube-proxy/app/server.go

```diff
@@ -79,8 +72,9 @@ import (
 )
 
 const (
-	proxyModeUserspace = "userspace"
-	proxyModeIPTables  = "iptables"
+	proxyModeUserspace   = "userspace"
```

Comment:
I don't like how non-obvious it is that these don't all work on any give platform.  Can we move these constants into the per-OS files and call a function from there to get a list of options to show in strings?  We should also deal with flags and componentconfig that do not map across OSes.

How much code reuse are we really getting by having these things live in the same sub-repo?  Would it be better to just make a win-kube-proxy ?  So much of what is left here is still linux-specific

Reply:
I would prefer a plugin model here, to provide interfaces for Service/Endpoint Updates, and just customize it in the plugin, rather than bringing a lot of code in to Kubernetes, along with the vendor changes.